### DISCUSSION TITLE
Feature Request: Supports subquery in lambda expression

### DISCUSSION THREAD

**Discussion by ritalin** [duckdb/duckdb#11742] [Ideas]
I have two tables.

> Table A
> 
> | Name  | Type |
> | ------------- | ------------- |
> | id  | numeric  |
> | name  | varchar  |

> Table B
>
> | Name  | Type |
> | ------------- | ------------- |
> | names  | varchar[]  |

Now, I'd like to convert names value of B into id of A.

For instance:

> Table A records
>
> | id  | name |
> | ------------- | ------------- |
> | 11 | 'foo'  |
> | 13 | 'var'  |
> | 17 | 'baz' |

> Table B records
> | names  | 
> | ------------- | 
> | ['foo', 'bar']  | 
> | ['foo', 'foo', 'baz']  | 

Converted results is:

> | ids  | 
> | ------------- | 
> | [11, 13]  | 
> | [11, 11, 17]  | 

I've tried to execute a following SQL:

```sql
select names.list_transform(s -> (select id from A where A.name = s)) from B
```

Currently, this query has failed with a following message.

> Invalid Input Error: Subqueries are not supported in lambda expressions!

**Comment by Tishj** [comment]
Please provide scripts when making issues like this.

```py
import duckdb

duckdb.execute("""
	create table A as FROM ( VALUES
		(11, 'foo'),
		(13, 'var'),
		(17, 'baz')
	) t(id, name)
""")

duckdb.execute("""
	create table B as FROM ( VALUES
		(['foo', 'bar']),
		(['foo', 'foo', 'baz'])
	) t(names)
""")

duckdb.sql("""
with mapping as (
   from A select map_from_entries(list((name, id))) name_to_id
)
from B, mapping
select names.apply(name -> name_to_id[name][1])
""").show()
```

```
┌──────────────────────────────────────────────────┐
│ "names".apply(("name" -> name_to_id["name"][1])) │
│                     int32[]                      │
├──────────────────────────────────────────────────┤
│ [11, NULL]                                       │
│ [11, 11, 17]                                     │
└──────────────────────────────────────────────────┘
```

**Reply by ritalin** [reply]
This is a nice workaround but curious about the creating cost of the lookup table.
Thank you.
